extras/src/main/java/com/novoda/espresso/AccessibilityServiceTogglingActivity.java

```diff
@@ -0,0 +1,45 @@
+package com.novoda.espresso;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.novoda.espresso.AccessibilityServiceToggler.Service;
+
+public class AccessibilityServiceTogglingActivity extends Activity {
+
+    public static final String ACTION_DISABLE_ALL_SERVICES = "com.novoda.espresso.DISABLE_ALL_SERVICES";
+    public static final String ACTION_SET_SERVICE = "com.novoda.espresso.SET_SERVICE";
+    public static final String ENABLED = "enabled";
+    public static final String DISABLED = "disabled";
+
+    private AccessibilityServiceToggler serviceToggler;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        serviceToggler = AccessibilityServiceToggler.create(getContentResolver());
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+
+        if (ACTION_SET_SERVICE.equalsIgnoreCase(action)) {
+            performAction(intent);
+        } else if (ACTION_DISABLE_ALL_SERVICES.equalsIgnoreCase(action)) {
+            serviceToggler.disableAll();
+        }
+
+        finish();
+    }
+
+    private void performAction(Intent intent) {
+        for (Service service : Service.values()) {
+            String value = intent.getStringExtra(service.name());
+            if (ENABLED.equalsIgnoreCase(value)) {
+                serviceToggler.enable(service);
```

Comment:
extracting the ServiceToggler lets us test it - especially with the more complex logic.

https://github.com/novoda/espresso-support/pull/16/files#diff-373a366c1b3ba51b0263938892bdd484L50 the previous version would disable ALL accessibility services when trying to disable just TalkBack.